src/vs/workbench/api/browser/mainThreadChatAgents2.ts

```diff
@@ -385,3 +386,37 @@ function computeCompletionRanges(model: ITextModel, position: Position, reg: Reg
 
 	return { insert, replace };
 }
+
+namespace ChatNotebookEdit {
+	export function fromChatEdit(part: IChatNotebookEditDto): IChatNotebookEdit {
+		return {
```

Comment:
There is data conversion and reviving in a few places (my other comments) and I think this is _the_ place. All data converting should happen here  this is the place where EH-data enters the renderer and after this it should be "correct"